wdn/templates_4.1/scripts/search.js

```diff
@@ -99,29 +106,78 @@ define(['jquery', 'wdn', 'require', 'modernizr', 'navigation'], function($, WDN,
 				});
 				domSearchForm.append(domEmbed);
 
+				var createSearchFrame = function() {
+					// lazy create the search iframe
+					if (!$unlSearch) {
+						$unlSearch = $('<iframe>', {
+							name: 'unlsearch',
+							id: 'wdn_search_frame',
+							title: 'Search results',
+							src: searchFrameAction
+						});
+
+						domSearchForm.parent().append($unlSearch).append($progress);
+
+						$unlSearch.on('load', function() {
+							postReady = true; // iframe should be ready to post messages to
+						});
+					}
+				};
+
+				var activateSearch = function() {
+					domSearchForm.parent().addClass('active');
+					$progress.show();
```

Comment:
`$progress.hide()` should now be added to the `close()` function. If you close the search before it finishes, a loading bar will be stuck on the screen.